src/main/java/io/reactivex/processors/UnicastProcessor.java

```diff
@@ -103,13 +120,44 @@
     }
 
     /**
+     * Creates an UnicastProcessor with the given internal buffer capacity hint, delay error flag and a callback for
+     * the case when the single Subscriber cancels its subscription.
+     *
+     * <p>The callback, if not null, is called exactly once and
+     * non-overlapped with any active replay.
+     *
+     * @param <T> the value type
+     * @param capacityHint the hint to size the internal unbounded buffer
+     * @param onCancelled the non null callback
+     * @param delayError deliver pending onNext events before onError
+     * @return an UnicastProcessor instance
+     * @since 2.0.8 - experimental
+     */
+    @CheckReturnValue
+    @Experimental
+    public static <T> UnicastProcessor<T> create(int capacityHint, Runnable onCancelled, boolean delayError) {
+        return new UnicastProcessor<T>(capacityHint, onCancelled,delayError);
+    }
+
+    /**
      * Creates an UnicastProcessor with the given capacity hint.
      * @param capacityHint the capacity hint for the internal, unbounded queue
      * @since 2.0
      */
     UnicastProcessor(int capacityHint) {
+        this(capacityHint, true);
+    }
+
+    /**
+     * Creates an UnicastProcessor with the given capacity hint and delay error flag.
+     * @param capacityHint the capacity hint for the internal, unbounded queue
+     * @param delayError deliver pending onNext events before onError
+     * @since 2.0.8 - experimental
+     */
+    UnicastProcessor(int capacityHint, boolean delayError) {
```

Comment:
3 args constructor has non-null onTerminate check; should i move this check to factory?